### DISCUSSION TITLE
Having issues with hydration directive and script setup

### DISCUSSION THREAD

**Discussion by rylanharper** [ElMassimo/iles#125] [Q&A] [answered]
Hey there everyone! I am currently using îles on a website I am building for a local wellness center. I am enjoying it a lot so far!

I am running into an issue within my mobile menu toggle that uses pinia; I believe the core of the issue stems from my lack of understanding using the `Hydration Directives` that îles offers. In my local environment it works wonderfully, but once I try and deploy it to Netlify I run into a build error: `Incorrect usage of hydration strategy in script setup`.

`components/nav-mobile.vue`:
```ts
<script setup lang="ts">
import { useStore } from '@/store'
const store = useStore()

function handleClick() {
  store.toggleMenu()
}
</script>

<template>
 // Extra code
  <button
    aria-label="Menu Toggle"
    @click="handleClick"
    @keyup.enter="handleClick"
  >
</template>
```

`store/index.ts`:
```ts
import { defineStore } from 'pinia'

export const useStore = defineStore('main', {
  state: () => {
    return {
      menuOpen: false
    }
  },
  actions: {
    toggleMenu(state?: boolean) {
      if (typeof state === 'boolean') {
        this.menuOpen = state
      } else {
        this.menuOpen = !this.menuOpen
      }
    }
  }
})
```

Here is my component with `client:visible`. My issue still persists:

`components/nav-mobile.vue`: 
```ts
<script setup client:visible lang="ts">
import { useStore } from '@/store'
const store = useStore()

function handleClick() {
  store.toggleMenu()
}
</script>

<template>
  // Extra code
  <button
    aria-label="Menu Toggle"
    @click="handleClick"
    @keyup.enter="handleClick"
  >
</template>
```

If anyone has any input for how I can effectively use my hydration directive so I can get my mobile-toggle/mobile-menu working on live that would be great! :)

**Comment by ElMassimo** [comment, marked as answer]
Hi Rylan,

This is something that the [_Client Scripts_](https://iles-docs.netlify.app/guide/client-scripts#client-script-block) doesn't mention explicitly, but anything that you define in a _Client Script_ is essentially separate from the Vue component, it will just be injected next to where it's rendered. It's an alternative to islands, where you need client-side code that does not map to a component.

The hydration directives should not be used in `<script setup>` but in a separate plain `script` tag, and you wouldn't be ableto share state or methods, since they are running in different contexts (`script setup` runs when building the site, the client script runs in the browser).

---

I'd recommend not using Pinia in client scripts, and instead simplify and use plain `ref` instead.

See [this example in the docs site](https://github.com/ElMassimo/iles/blob/90a5948853111a71ffd8019d4624c6d9c9f620dc/docs/src/components/TheSidebar.vue#L1-L30), which is very similar to what you are trying to achieve here 😃 

Also see [this thread](https://github.com/ElMassimo/iles/discussions/70#discussioncomment-2362578) for a similar discussion with different approaches, including some that don't even use Vue.
